piptools/utils.py

```diff
@@ -441,3 +442,41 @@ def get_sys_path_for_python_executable(python_executable: str) -> List[str]:
     assert isinstance(paths, list)
     assert all(isinstance(i, str) for i in paths)
     return [os.path.abspath(path) for path in paths]
+
+
+def omit_list_value(lst: List[_T], value: _T) -> List[_T]:
+    """Produce a new list with a given value skipped."""
+    return [item for item in lst if item != value]
+
+
+_strip_extras_re = re.compile(r"\[.+?\]")
+
+
+def strip_extras(name: str) -> str:
+    """Strip extras from package name, e.g. pytest[testing] -> pytest."""
+    return re.sub(_strip_extras_re, "", name)
+
+
+def copy_install_requirement(template: InstallRequirement) -> InstallRequirement:
+    """Make a copy of an ``InstallRequirement``."""
```

Comment:
Thoughts on `combine_install_requirements` delegating some of its logic to this new function?

Reply:
Agreed. Would you like to open a PR against my `new-resolver` branch I'd be happy to merge?

Reply:
Sure thing. Not sure if I'll make it in time for your PR's merge, but it can be a separate cleanup, if so.

Reply:
Addressed in c8aca929bbd7d6c49907a445ed15f7e7edee18ee